neonvm-runner/cmd/main.go

```diff
@@ -625,9 +635,10 @@ func newConfig(logger *zap.Logger) *Config {
 		appendKernelCmdline:  "",
 		skipCgroupManagement: false,
 		diskCacheSettings:    "cache=none",
-		memoryProvider:       "", // Require that this is explicitly set. We'll check later.
-		autoMovableRatio:     "", // Require that this is explicitly set IFF memoryProvider is VirtioMem. We'll check later.
-		cpuScalingMode:       "", // Require that this is explicitly set. We'll check later.
+		memoryProvider:       "",
+		autoMovableRatio:     "",
+		cpuScalingMode:       "",
+		architecture:         runtime.GOARCH,
```

Comment:
question: Is this only ever equal to `runtime.GOARCH` ? Do we plan to keep it that way? (if so: can we make assumptions that e.g. we're only ever executing `qemu-system-x86_64` from `amd64` go code?)

Reply:
Yes, and I think I got where are you heading to - why not to use `runtime.GOARCH` everywhere?

Reply:
>But also — why install both qemu-system-x86_64 and qemu-system-aarch64 if we're only going to need one? (and then if we like, we can set up some symlink to refer to it, so we don't need to match the name inside the go code)



Basically to simplify the image builds, since this PR was introduced way before crossplatform images. I can update PR with crossplatform images to use only the correct version of the qemu.

But in general, cfg.architecture is used in a couple of places (kernel arguments, qemu arguments (not binary name), adding firmware) and I don't want to make it using `runtime.GOARCH` everywhere instead. 

so: 
- I will update the PR with #1138 to have only one qemu binary
- I keep cfg.architecture here, for the sake of using it in different places

How does it sound?

Reply:
Fair enough, sounds like a plan :)